### DISCUSSION TITLE
How to create an installable release?

### DISCUSSION THREAD

**Discussion by TheAutomaTom** [cawa-93/vite-electron-builder#919] [Q&A]
Howdy y'all,
  _I am really enjoying learning some new things about CI and such with your great template!_  May I ask on clarification of the release workflow?  I have all ci.yalm1 tests passing.  When I use github to create a release... the product is just my source code.  Should I be setting something in `.idea\deployment.xml`?  
  _I was expecting a build folder to be created on my local drive, but the CI workflow is an awesome touch!_
Thanks you for any advice.  I hope I might be able to flush out the docs to help others, if I can get some help, here.

**Comment by cawa-93** [comment]
[.idea](https://github.com/cawa-93/vite-electron-builder/tree/main/.idea) contain IDE configs only. This has nothing to do with the build. If you do not use the IDE from jetbrains, you are free to delete this directory entirely.

- `npm run build` — bundle local assets (transpile ts to js, transpile css, bundled modules, typical bundler stuff)
- `npx electron-builder --config .electron-builder.config.js` — creating a binary program from the bundled assets

Check https://github.com/cawa-93/vite-electron-builder/blob/main/.github/workflows/release.yml. It contain full _production_ pipeline including building assets and compilation binaries. Basically to get installers on your local machine you should run `npm run build && npx electron-builder --config .electron-builder.config.js`.

Here are also [`npm run compile`](https://github.com/cawa-93/vite-electron-builder/blob/c0efcb17090a2ee0b8c0db09669beaea33ade605/package.json#L17) script. It create binnaries but it is _non-production_. This script is intended to create a "production-like" binary for further debugging locally, not for distribution to end users.

[`npm run compile`](https://github.com/cawa-93/vite-electron-builder/blob/c0efcb17090a2ee0b8c0db09669beaea33ade605/package.json#L17) Basically is just `npm run build && npx electron-builder --config .electron-builder.config.js` but with extra configs for local dev improvements.

**Reply by TheAutomaTom** [reply]
Thanks for that help!

**Reply by lanseria** [reply]
I encountered while using GitHub Actions release feature. Although the actions show success, no release files were actually published. I am wondering what could be the reason behind this.

Could you please provide some insights on how to solve this issue?

My repository url: https://github.com/lanseria/test-electron-mapbox/actions
